### ISSUE TITLE
Prev/Next button not working with visibleRange example

### ISSUE THREAD

**Issue by linonetwo** [fullcalendar/fullcalendar#7129]
### Reduced Test Case

https://codepen.io/linonetwo/pen/zYLNGNP?editors=001

### Do you understand that if a reduced test case is not provided, we will intentionally delay triaging of your ticket?

- [X] I understand

### Which connector are you using (React/Angular/etc)?

No connector (vanilla JS)

### Bug Description

Simillar to https://github.com/fullcalendar/fullcalendar/issues/4386#issuecomment-1374503062

https://user-images.githubusercontent.com/3746270/211156369-1289882a-d071-4b68-a596-d2547215e7ae.mov

You can't go to future any more.

**Comment by acerix** [comment]
Thanks, confirmed.

I tried the more simple example, removing the times from start/end, and with `dateIncrement` set but it still doesn't work properly:

https://codepen.io/acerix/pen/oNMGNpE?editors=001

The examples in the docs initially set `endDate` with the current time of day which makes that day included, but after the date changed, the time becomes '00:00:00' so the last day is no longer included.

As a workaround, you can use custom buttons instead of the built-in ones, and manually change the date with `incrementDate`.

**Comment by linonetwo** [comment]
But How to access calendar instance to call `incrementDate`?

```js
  customButtons: {
    myCustomButton: {
      text: 'custom!',
      click: function() {
        alert('clicked the custom button!');
      }
    }
  },
```

**Comment by pathaugen** [comment]
Was experiencing the same issue and found this open issue with a pull request to resolve it.

Don't have the permissions to check the conflicts and resolve, but the code to fix this issue should we resolve conflicts or take into the current branch and re-create a pull request?

**Comment by linonetwo** [comment]
Of course, you can see I've been waiting for a fix for 3 years. I have updated the branch, so there are no conflicts. Hope the official team can take time to review and merge it.

@arshaw @acerix Could you please use PR to do major update or refactor? So there won't be so many conflicts, and it will be easier to rebase (on a single PR). I can see you are upgrading a lot of things in several small commits, which is hard to track and follow.

While I'm glad to see you are still actively updating this project.
